internal/store/parent_subnet.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package store
+
+import (
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/mattermost/genesis/model"
+	"github.com/pkg/errors"
+)
+
+var parentSubnetSelect sq.SelectBuilder
+
+func init() {
+	parentSubnetSelect = sq.
+		Select("ParentSubnet.ID", "CIDR", "SplitRange", "CreateAt",
+			"LockAcquiredBy", "LockAcquiredAt").
+		From("ParentSubnet")
+}
+
+type rawParentSubnet struct {
+	*model.ParentSubnet
+}
+type rawParentSubnets []*rawParentSubnet
+
+func (r *rawParentSubnet) toParentSubnet() (*model.ParentSubnet, error) {
+	return r.ParentSubnet, nil
+}
+
+func (rc *rawParentSubnets) toParentSubnets() ([]*model.ParentSubnet, error) {
+	var parentSubnets []*model.ParentSubnet
+	for _, rawParentSubnet := range *rc {
+		parentSubnet, err := rawParentSubnet.toParentSubnet()
+		if err != nil {
+			return nil, err
+		}
+		parentSubnets = append(parentSubnets, parentSubnet)
+	}
+
+	return parentSubnets, nil
+}
+
+// GetParentSubnet fetches the given parent subnet by subnet range.
+func (sqlStore *SQLStore) GetParentSubnet(id string) (*model.ParentSubnet, error) {
```

Comment:
I think `model.ParentSubnet` it's relatively small object as I see and you can return it without a pointer in the function.

Reply:
Great suggestion, thanks. Updated everywhere :)